File: Bridge/Vasylyshyn/src/Shape/Shape.java
package Shape;

import Color.Color;

public abstract class Shape {
    protected Color color;
    public abstract double Area();
    public abstract double Perimeter();
    public abstract void draw();

    public Shape() {
    }

    public Shape(Color color) {

        this.color = color;
    }

    public Color getColor() {
        return color;
    }

    public Shape setColor(Color color) {
        this.color = color;
        return this;
    }




    @Override
    public String toString() {
        return "Shape{" +
                ", color=" + color +
                '}';
    }
}

File: Bridge/Severylova-03/Bridge/Bridge/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bridge
{
    abstract class Shape
    {
        protected Color _color;
        public Shape(Color color) 
        {
            this._color = color;
        }
        public abstract double Perimeter();


        public abstract double Area();

        public override string ToString()
        {
            return $"Shape: {GetType().Name}, Color: {_color.GetColor()}, Area: {Area()}, Perimeter: {Perimeter()}";
        }
    }

    
}
